arrow-libs/plugins/src/main/kotlin/arrow/gradle/MppPublishingPlugin.kt

```diff
@@ -0,0 +1,159 @@
+package arrow.gradle
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.UnknownTaskException
+import org.gradle.api.artifacts.dsl.RepositoryHandler
+import org.gradle.api.artifacts.repositories.MavenArtifactRepository
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.jvm.tasks.Jar
+import org.gradle.kotlin.dsl.named
+import org.gradle.kotlin.dsl.register
+import org.gradle.kotlin.dsl.withType
+import org.gradle.plugins.signing.SigningExtension
+
+/**
+ * An empty [Plugin] that exposes a set of utility functions to simplify setting up MPP publishing.
+ */
+class MppPublishingPlugin : Plugin<Project> {
+  override fun apply(target: Project) {}
+}
+
+fun Project.setupPublishing(
+  pomDevId: String,
+  pomDevName: String,
+  projectUrl: String,
+  projectDesc: String,
+  releaseRepo: java.net.URI,
+  snapshotRepo: java.net.URI,
+  sonatypeUsername: String? = System.getenv("SONATYPE_USER"),
+  sonatypePassword: String? = System.getenv("SONATYPE_PWD"),
+  pass: String? = System.getenv("SIGNINGPASSWORD"),
+  key: String? = System.getenv("SIGNINGKEY"),
+  licenseName: String = "The Apache Software License, Version 2.0",
+  licenseUrl: String = "https://www.apache.org/licenses/LICENSE-2.0.txt",
+  licenseDist: String = "repo",
+  scmConnection: String = "${projectUrl.replaceFirst("https://", "scm:git:git://")}.git",
+  scmDevConnection: String = "${projectUrl.replaceFirst("https://", "scm:git:ssh://")}.git",
+) {
+  afterEvaluate {
+    extensions.getByType(PublishingExtension::class.java).apply {
+      val mavenPublications = publications.withType<MavenPublication>()
+      mavenPublications.all {
+        artifact(project.javadocJar())
+        setupPom(
+          url = projectUrl,
+          description = projectDesc,
+          pomDevId = pomDevId,
+          pomDevName = pomDevName,
+          licenseName = licenseName,
+          licenseUrl = licenseUrl,
+          licenseDist = licenseDist,
+          scmConnection = scmConnection,
+          scmDevConnection = scmDevConnection
+        )
+      }
+      repositories {
+        maven(
+          if (version.toString().endsWith("SNAPSHOT")) snapshotRepo else releaseRepo,
+          sonatypeUsername,
+          sonatypePassword
+        )
+      }
+      Nullable.zip(key, pass) { key, pass -> signPublications(key, pass) }
```

Comment:
I love the `zip` function, however, I wonder if we could do this too:

```kotlin
if (key != null && pass != null) signPublications(key, pass)
```

They are, to all effects, the same :thinking: 

Or add Arrow itself to the plugin classpath :joy:   
PS: Not a blocker

Reply:
Ye, I was thinking that too. Using `!= null` is probably easier here.

Let's not put Arrow on the classpath 😂 It's hard enough to work with without more new things.